DuckDuckGo/TabViewController.swift

```diff
@@ -1375,55 +1375,64 @@ extension TabViewController: WKNavigationDelegate {
             NotificationCenter.default.post(Notification(name: AppUserDefaults.Notifications.didVerifyInternalUser))
         }
 
-        // Important: Order of these checks matter!
-        if urlSchemeType == .blob {
-            // 1. To properly handle BLOB we need to trigger its download, if temporaryDownloadForPreviewedFile is set we allow its load in the web view
-            if let temporaryDownloadForPreviewedFile, temporaryDownloadForPreviewedFile.url == navigationResponse.response.url {
-                // BLOB already has a temporary downloaded so and we can allow loading it
-                blobDownloadTargetFrame = nil
-                decisionHandler(.allow)
-            } else {
-                // First we need to trigger download to handle it then in webView:navigationAction:didBecomeDownload
-                decisionHandler(.download)
-            }
-        } else if FilePreviewHelper.canAutoPreviewMIMEType(mimeType) {
-            // 2. For this MIME type we are able to provide a better custom preview via FilePreviewHelper so it takes priority
-            let download = self.startDownload(with: navigationResponse, decisionHandler: decisionHandler)
-            mostRecentAutoPreviewDownloadID = download?.id
-            Pixel.fire(pixel: .downloadStarted,
-                       withAdditionalParameters: [PixelParameters.canAutoPreviewMIMEType: "1"])
-        } else if shouldTriggerDownloadAction(for: navigationResponse),
-                  let downloadMetadata = AppDependencyProvider.shared.downloadManager.downloadMetaData(for: navigationResponse.response) {
-            // 3a. We know it is a download, but allow WebKit handle the "data" scheme natively
-            if urlNavigationalScheme == .data {
-                decisionHandler(.download)
-                return
-            }
-
-            // 3b. We know the response should trigger the file download prompt
-            self.presentSaveToDownloadsAlert(with: downloadMetadata) {
-                self.startDownload(with: navigationResponse, decisionHandler: decisionHandler)
-            } cancelHandler: {
+        // If the navigation has been handled by the special error page handler cancel navigating to new content as the special error page will be shown.
+        Task { @MainActor in
```

Comment:
Whoa :), this is triggering an alert for me. Making a task on this code could have bigger consequences – we’re essentially returning immediately and skipping the run loop, without a guarantee that other tasks spawned elsewhere won’t interfere with this. Are we sure this is the approach we want to take?